modules/library/jdbc/src/test/java/org/geotools/jdbc/JDBCPrimaryKeyFinderViewTestSetup.java

```diff
@@ -0,0 +1,70 @@
+package org.geotools.jdbc;
+
+import java.sql.SQLException;
+
+public abstract class JDBCPrimaryKeyFinderViewTestSetup extends JDBCDelegatingTestSetup {
+
+    protected JDBCPrimaryKeyFinderViewTestSetup(JDBCTestSetup delegate) {
+        super(delegate);
+    }
+
+    @Override
+    protected final void setUpData() throws Exception {
+        // kill all the data
+        try {
+            dropMetadataView();
+        } catch (SQLException e) {
+        }
+        try {
+            dropMetadataTable();
+        } catch (SQLException e) {
+        }
+        try {
+            dropSequencedPrimaryKeyTable();
+        } catch (SQLException e) {
+        }
+
+        // create all the data
+        createMetadataTable();
+        createSequencedPrimaryKeyTable();
+        createMetadataView();
+    }
+
+    /** Drops the metadata table */
+    protected abstract void dropMetadataTable() throws Exception;
+
+    /** Drops the metadata view */
+    protected abstract void dropMetadataView() throws Exception;
+
+    /** Drops the "seqtable" table. */
+    protected abstract void dropSequencedPrimaryKeyTable() throws Exception;
+
+    /**
+     * Creates the default geotools metadata table, empty.
+     *
+     * <p>The table is named "gt_pk_metdata". See {@link MetadataTablePrimaryKeyFinder} javadoc for
```

Comment:
By the looks of the implementations, this would be a better javadoc:

```suggestion
     * <p>The table is named "tbl_gt_pk_metdata". See {@link MetadataTablePrimaryKeyFinder} javadoc for
```

Reply:
Done.